includes/blocks/course-list/class-sensei-course-list-block.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * File containing extra functionalities of extended Sensei_Course_List_Block class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Sensei_Course_List_Block
+ */
+class Sensei_Course_List_Block {
+
+	/**
+	 * Sensei_Course_List_Block constructor.
+	 */
+	public function __construct() {
+		add_filter( 'render_block', [ $this, 'render_login_form_if_applicable' ], 10, 2 );
+	}
+
+	/**
+	 * Replaces content of Course List block when Show Login Form is set and user is logged out.
+	 *
+	 * @param string $block_content The block content to be rendered.
+	 * @param array  $block          The block to be rendered.
+	 *
+	 * @return string
+	 */
+	public function render_login_form_if_applicable( $block_content, $block ) {
```

Comment:
We should mark this as `@private`. You also usually see functions like this using a naming convention like `maybe_render_login_form`.

Reply:
Absolutely, updated the name and annotation [here](https://github.com/Automattic/sensei/pull/6343/commits/bd70cd1c70701b7e7fe5c632d80e2de2a7b738a2)